File: test/functions/python/mongo.py
from pymongo import MongoClient
import os
import swifty

def main(req):
    db = swifty.MongoDatabase(req.args['dbname'])
    colname = req.args['collection']
    col = db[colname]

    if req.args['action'] == 'insert':
        col.insert_one({ "key": req.args['key'], "val": req.args['val'] })
        return { "res": "done" }

    if req.args['action'] == 'select':
        res = col.find_one({ "key": req.args['key'] })
        return { "res": res['val'] }

    return { "res": "invalid" }, None

File: test/functions/golang/mongo.go
package main

import (
	"fmt"
	"swifty"
	"gopkg.in/mgo.v2/bson"
)

func Main(rq *Request) (interface{}, *Responce) {
	db, err := swifty.MongoDatabase(rq.Args["dbname"])
	if err != nil {
		fmt.Println(err)
		panic("Can't get mgo dbase")
	}

	col := db.C(rq.Args["collection"])
	res := "done"

	switch rq.Args["action"] {
	case "insert":
		err = col.Insert(bson.M{"key": rq.Args["key"], "val": rq.Args["val"]})
	case "select":
		var ent struct { Key, Val string }

		err = col.Find(bson.M{"key": rq.Args["key"]}).One(&ent)
		if err == nil {
			res = ent.Val
		}
	default:
		res = "invalid"
	}

	if err != nil {
		res = "operation failed"
	}

	return map[string]string{"res": res}, nil
}
